### PR TITLE
rule: add `union_now_doc_hidden`

### PR DESCRIPTION
**Pull request by vasucp1207**
Part of https://github.com/obi1kenobi/cargo-semver-checks/issues/578

- [x] More tests

### CODE REVIEW

File: test_crates/union_now_doc_hidden/new/cargo.toml
Comment (obi1kenobi) [inline test_crates/union_now_doc_hidden/new/cargo.toml:1]: I've never seen `cargo.toml` as the filename for this, I've only ever seen `Cargo.toml` (note the capitalization difference).

I'm curious how you generated this file — did you use the scripts for creating new lints and test files? Are you using a filesystem that lowercases everything?

**Review by obi1kenobi — COMMENTED** [review]
So far this looks good! Let me know if you get stuck on anything and if you'd like me to take a closer look.

File: test_crates/union_now_doc_hidden/new/cargo.toml
Comment (vasucp1207) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: For some reason `Cargo.toml` is not generated after running the script but all other files do. My bad I named it wrongly.

Comment (obi1kenobi) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: Oh interesting, that sounds like a bug in the script that we should fix. Which script did you have this experience with?

Comment (vasucp1207) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: Running `./scripts/regenerate_test_rustdocs.sh` also gives the error,
`cargo-semver-checks/scripts/regenerate_test_rustdocs.sh: line 23: shopt: globstar: invalid shell option name`

Comment (vasucp1207) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: After commenting out line no 23 it works fine.

**Comment by pksunkara** [comment]
Looks like this lint can be unified too with `enum` and `struct`. Not a blocker for this PR though.

File: test_crates/union_now_doc_hidden/new/cargo.toml
Comment (vasucp1207) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: > Oh interesting, that sounds like a bug in the script that we should fix. Which script did you have this experience with?

make_new_lint.sh

**Comment by obi1kenobi** [comment]
> Looks like this lint can be unified too with `enum` and `struct`. Not a blocker for this PR though.

Yeah, lots of lints can be unified like that. At this stage that's predominantly a cosmetic improvement, though, so having the lint in any form is better than trying to squeeze everything into as few lints as possible.

**Review by obi1kenobi — APPROVED** [review]
Nicely done! 🚀

File: test_crates/union_now_doc_hidden/new/cargo.toml
Comment (obi1kenobi) [reply test_crates/union_now_doc_hidden/new/cargo.toml:1]: Opened #680 for this, feel free to tackle it if you find it interesting! Our tooling gets better one small improvement at a time — it really adds up over the course of weeks and months.

**Pull request merged**